# mayhem_program::instructions::pump_interface::sell::handler  (yours 7480 B at .text+0x1dfb8; original 12992 B at .text+0x1f100)
# distance 1795 instructions; evidence: called-by mayhem_program::__private::__global::sell#c304; same-callees 20/25
--- original
+++ yours
@@ -1,212 +1,297 @@
-mov64 r8, r5
 mov64 r7, r4
-stxdw [r10-1152], r3
-stxdw [r10-1144], r2
-mov64 r9, r1
-ldxdw r6, [r9+8]
-ldxdw r2, [r6+8]
-ldxdw r1, [r6+16]
-add64 r1, 8
-add64 r2, 8
-call orig_fn_2dbf8
-jne r0, 2, +0x908
-ldxdw r1, [r6+72]
-ldxdw r2, [r1+96]
-ldxdw r3, [r6+8]
-ldxdw r1, [r6+16]
-stdw [r10-1112], 0
-stdw [r10-1120], 0
-stdw [r10-1128], 0
-stdw [r10-1136], 0
-add64 r1, 8
-add64 r3, 8
-mov64 r4, r10
-add64 r4, -1136
-call orig_fn_2e0a0
-jne r0, 2, +0x908
-mov64 r1, 2
-stxdw [r10-1176], r1
-ldxdw r1, [r6+72]
-stxdw [r10-1184], r1
-ldxdw r1, [r1+72]
-stxdw [r10-1168], r1
-jeq r1, 0, +0x7a8
-stxdw [r10-1192], r9
-stxdw [r10-1216], r8
-stxdw [r10-1200], r6
+mov64 r8, r3
+mov64 r6, r2
+ldxdw r9, [r1+8]
+ldxdw r2, [r9+8]
+ldxb r2, [r2+253]
+jeq r2, 0, +0xc8
+mov64 r1, r10
+add64 r1, -1000
+lddw r2, .rodata[00000000]
+call mayhem_program::MayhemError::name#2984
+mov64 r1, r10
+add64 r1, -976
+mov64 r2, 0
+call <T as alloc::string::SpecToString>::spec_to_string#2731
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1024], r1
+stb [r10-952], 2
+stw [r10-1008], 161
+stdw [r10-1016], 58
+stdw [r10-1032], 0
+stw [r10-880], 6000
+ja +0x418
+stxdw [r10-2520], r5
+stxdw [r10-2528], r1
+mov64 r1, r10
+add64 r1, -1032
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-1032]
+jne r1, 0, +0x1b60
+stxdw [r10-2512], r6
+ldxdw r1, [r10-992]
+ldxdw r2, [r9+16]
+ldxdw r3, [r2+88]
+jeq r3, 0, +0x228
+ldxdw r4, [r9+8]
+ldxdw r6, [r4+237]
+jeq r6, 0, +0x228
+mov64 r0, 1
+mov64 r4, 1
+jslt r6, 0, +0x340
+mov64 r4, 0
+mov64 r5, r3
+add64 r5, r6
+jsge r5, r3, +0x358
+xor64 r4, r0
+and64 r4, 1
+jne r4, 0, +0x378
+jsge r1, r5, +0x228
+mov64 r1, r10
+add64 r1, -1000
+lddw r2, .rodata[0f000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -976
+mov64 r2, 15
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1024], r1
+stb [r10-952], 2
+stw [r10-1008], 165
+stdw [r10-1016], 58
+stdw [r10-1032], 0
+stw [r10-880], 6015
+ja +0x418
+stxdw [r2+88], r1
+mov64 r1, r10
+add64 r1, -1032
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-1032]
+jne r1, 0, +0x1bc0
+ldxdw r2, [r10-992]
+ldxdw r1, [r9+16]
+ldxdw r3, [r1+80]
+jsge r2, r3, +0x390
+ldxb r2, [r1+104]
+jeq r2, 1, +0x440
+ldxdw r2, [r9+8]
+ldxdw r2, [r2+144]
+ldxdw r3, [r9+72]
+ldxdw r3, [r3+96]
+jge r3, r2, +0x438
+mov64 r1, r10
+add64 r1, -1000
+lddw r2, .rodata[030000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -976
+mov64 r2, 3
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1024], r1
+stb [r10-952], 2
+stw [r10-1008], 169
+stdw [r10-1016], 58
+stdw [r10-1032], 0
+stw [r10-880], 6003
+ja +0x418
+mov64 r5, r3
+add64 r5, r6
+jslt r5, r3, +0x178
+mov64 r0, 0
+xor64 r4, r0
+and64 r4, 1
+jeq r4, 0, +0x190
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:165:41)
+call core::panicking::panic_const::panic_const_add_overflow#3493
+mov64 r1, r10
+add64 r1, -1000
+lddw r2, .rodata[04000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -976
+mov64 r2, 4
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1024], r1
+stb [r10-952], 2
+stw [r10-1008], 167
+stdw [r10-1016], 58
+stdw [r10-1032], 0
+stw [r10-880], 6004
+mov64 r1, r10
+add64 r1, -1032
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+exit
+stb [r1+104], 1
+ldxdw r2, [r9+72]
+ldxdw r1, [r2+72]
+stxdw [r10-2536], r1
+jeq r1, 0, +0x1460
+stxdw [r10-2552], r9
+stxdw [r10-2560], r2
+ldxdw r1, [r2+80]
+stxdw [r10-2544], r1
 and64 r7, 65535
-ldxdw r1, [r10-1152]
+mov64 r1, r8
 mov64 r2, 0
 mov64 r3, r7
 mov64 r4, 0
 call __multi3
 mov64 r9, r0
 mov64 r6, r1
-ldxdw r1, [r10-1144]
+ldxdw r1, [r10-2512]
 mov64 r2, 0
 mov64 r3, r7
 mov64 r4, 0
 call __multi3
+mov64 r2, r1
+add64 r2, r9
+mov64 r4, 1
+mov64 r3, 1
+jne r6, 0, +0x538
+mov64 r3, 0
+jge r2, r1, +0x540
+or64 r3, r4
+jne r3, 1, +0x558
+ja +0x1460
+jlt r2, r1, +0x520
+mov64 r4, 0
+or64 r3, r4
+jeq r3, 1, +0x1460
+mov64 r1, r0
+mov64 r3, 10000
+mov64 r4, 0
+call __udivti3
 mov64 r7, r0
-mov64 r8, r1
-add64 r8, r9
-mov64 r2, 1
-mov64 r9, 1
-jne r6, 0, +0x1c0
-mov64 r9, 0
-jlt r8, r1, +0x1d0
-mov64 r2, 0
-or64 r9, r2
-ldxdw r1, [r10-1184]
-ldxdw r1, [r1+80]
-stxdw [r10-1224], r1
+mov64 r9, r1
+mov64 r1, 1
+mov64 r6, 1
+ldxdw r2, [r10-2512]
+jge r2, r7, +0x6a8
+mov64 r6, 0
+jlt r8, r9, +0x6b0
+jeq r8, r9, +0x5c8
+mov64 r6, r1
+ldxdw r1, [r10-2544]
 mov64 r2, 0
 lddw r3, 0x38d7ea4c68000
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
 mov64 r1, r0
-ldxdw r3, [r10-1168]
+ldxdw r3, [r10-2536]
 mov64 r4, 0
 call __udivti3
-stxdw [r10-1208], r0
-stxdw [r10-1160], r1
-and64 r9, 1
-jne r9, 0, +0x7a8
-mov64 r1, r7
-mov64 r2, r8
-mov64 r3, 10000
-mov64 r4, 0
-call __udivti3
+and64 r6, 1
+jeq r6, 0, +0x1460
+ldxdw r6, [r10-2512]
+mov64 r4, r6
+add64 r4, r7
 mov64 r3, 1
 mov64 r2, 1
-ldxdw r4, [r10-1144]
-jge r4, r0, +0x2e8
-ldxdw r8, [r10-1200]
-ldxdw r4, [r10-1192]
-ldxdw r7, [r10-1152]
-jge r7, r1, +0x310
-jne r7, r1, +0x320
-and64 r2, 1
-jeq r2, 0, +0x338
-ja +0x7a8
-mov64 r2, 0
-ldxdw r8, [r10-1200]
-ldxdw r4, [r10-1192]
-ldxdw r7, [r10-1152]
-jlt r7, r1, +0x2c8
+jlt r4, r6, +0x6c8
+mov64 r2, 0
+mov64 r4, r8
+add64 r4, r9
+add64 r4, r2
+jge r4, r8, +0x6e8
+jeq r4, r8, +0x6f8
+mov64 r2, r3
+jne r2, 1, +0x700
+ja +0x1460
+jge r8, r9, +0x5b8
+mov64 r1, 0
+jne r8, r9, +0x5c0
+ja +0x5c8
+mov64 r4, r8
+add64 r4, r9
+add64 r4, r2
+jlt r4, r8, +0x688
 mov64 r3, 0
-jeq r7, r1, +0x2d0
+jne r4, r8, +0x690
+jeq r2, 1, +0x1460
+mov64 r3, r7
+mov64 r2, 1
+mov64 r4, 1
+jlt r6, r3, +0x728
+mov64 r4, 0
+mov64 r5, r8
+sub64 r5, r9
+sub64 r5, r4
+sub64 r6, r7
+mov64 r4, 1
+jlt r0, r6, +0x7b0
+mov64 r4, 0
+mov64 r6, 1
+jge r1, r5, +0x7c0
+jeq r1, r5, +0x7d0
+mov64 r4, r6
+ldxdw r5, [r10-2512]
+add64 r7, r5
+jge r7, r3, +0x7e8
+and64 r4, 1
+jeq r4, 0, +0x800
+ja +0x858
+mov64 r6, 1
+jlt r1, r5, +0x770
+mov64 r6, 0
+jne r1, r5, +0x778
+ldxdw r5, [r10-2512]
+add64 r7, r5
+jlt r7, r3, +0x798
+mov64 r2, 0
+and64 r4, 1
+jne r4, 0, +0x858
+add64 r9, r8
+add64 r9, r2
+mov64 r3, 1
+mov64 r2, 1
+jle r0, r7, +0x870
+mov64 r2, 0
+jgt r1, r9, +0x878
+jeq r1, r9, +0x888
 mov64 r2, r3
 and64 r2, 1
-jne r2, 0, +0x7a8
-ldxdw r5, [r10-1144]
-mov64 r4, r5
-add64 r4, r0
-mov64 r3, 1
-mov64 r2, 1
-jge r4, r5, +0x3a8
-mov64 r4, r7
-add64 r4, r1
-add64 r4, r2
-jge r4, r7, +0x3d0
-jne r4, r7, +0x3e0
+jne r2, 0, +0x898
+mov64 r1, 12
+call mayhem_program::instructions::pump_interface::sell::<impl core::convert::From<mayhem_program::instructions::pump_interface::sell::SellError> for anchor_lang::error::Error>::from#5915
+ja +0x430
+jle r1, r9, +0x838
+mov64 r3, 0
+jne r1, r9, +0x840
 and64 r2, 1
-jeq r2, 0, +0x3f8
-ja +0x7a8
-mov64 r2, 0
-mov64 r4, r7
-add64 r4, r1
-add64 r4, r2
-jlt r4, r7, +0x388
-mov64 r3, 0
-jeq r4, r7, +0x390
-mov64 r2, r3
-and64 r2, 1
-jne r2, 0, +0x7a8
-mov64 r3, r0
-mov64 r2, 1
-mov64 r4, 1
-ldxdw r5, [r10-1144]
-jlt r5, r3, +0x428
-mov64 r4, 0
-mov64 r5, r7
-sub64 r5, r1
-sub64 r5, r4
-ldxdw r6, [r10-1144]
-sub64 r6, r0
-mov64 r4, 1
-ldxdw r9, [r10-1208]
-jge r9, r6, +0x4d0
-mov64 r6, 1
-ldxdw r9, [r10-1160]
-jge r9, r5, +0x4f0
-ldxdw r9, [r10-1160]
-jne r9, r5, +0x508
-ldxdw r5, [r10-1144]
-add64 r0, r5
-jge r0, r3, +0x528
-mov64 r3, 12
-stxdw [r10-1176], r3
-and64 r4, 1
-jeq r4, 0, +0x550
-ja +0x7a8
-mov64 r4, 0
-mov64 r6, 1
-ldxdw r9, [r10-1160]
-jlt r9, r5, +0x480
-mov64 r6, 0
-ldxdw r9, [r10-1160]
-jeq r9, r5, +0x490
-mov64 r4, r6
-ldxdw r5, [r10-1144]
-add64 r0, r5
-jlt r0, r3, +0x4a8
-mov64 r2, 0
-mov64 r3, 12
-stxdw [r10-1176], r3
-and64 r4, 1
-jne r4, 0, +0x7a8
-add64 r1, r7
-add64 r1, r2
-mov64 r3, 1
-mov64 r2, 1
-ldxdw r4, [r10-1208]
-jgt r4, r0, +0x760
-ldxdw r4, [r10-1160]
-jgt r4, r1, +0x778
-ldxdw r4, [r10-1160]
-jne r4, r1, +0x790
-and64 r2, 1
-jeq r2, 0, +0x7a8
-ldxdw r1, [r10-1216]
+jeq r2, 0, +0x858
+ldxdw r1, [r10-2520]
 and64 r1, 65535
-jge r1, 10001, +0x3278
-ldxdw r7, [r8+8]
-ldxdw r1, [r8+40]
+jgt r1, 10000, +0x1cd8
+ldxdw r7, [r10-2552]
+ldxdw r1, [r7+40]
 ldxdw r6, [r1+104]
-ldxdw r1, [r10-1224]
-mov64 r2, 0
-mov64 r3, r6
+mov64 r1, r6
+mov64 r2, 0
+ldxdw r9, [r10-2544]
+mov64 r3, r9
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
 mov64 r3, r6
-ldxdw r1, [r10-1168]
+ldxdw r1, [r10-2536]
 add64 r3, r1
 mov64 r4, 1
-jlt r3, r6, +0x640
+jlt r3, r6, +0x930
 mov64 r4, 0
 mov64 r1, r0
 call __udivti3
-mov64 r6, r0
-jne r1, 0, +0x31d8
+mov64 r8, r0
+jne r1, 0, +0x1c20
 mov64 r3, 10000
-ldxdw r1, [r10-1216]
+ldxdw r1, [r10-2520]
 sub64 r3, r1
+ldxdw r1, [r10-2560]
+ldxdw r1, [r1+96]
 and64 r3, 65535
-ldxdw r1, [r10-1184]
-ldxdw r1, [r1+96]
 mov64 r2, 0
 mov64 r4, 0
 call __multi3
@@ -215,1281 +300,496 @@
 mov64 r3, 10000
 mov64 r4, 0
 call __udivti3
-ldxdw r1, [r7+104]
+ldxdw r1, [r7+8]
+ldxdw r1, [r1+112]
 mov64 r2, r1
-jge r1, r6, +0x910
-ldxdw r7, [r10-1224]
-jne r1, 0, +0x928
-jge r6, r0, +0x938
-jgt r7, r6, +0x948
-stw [r10-1136], 9
-mov64 r3, r10
-add64 r3, -1136
-lddw r1, str("called `Result::unwrap()` on an `Err` value")
-mov64 r2, 43
-lddw r4, table[3:<mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Debug>::fmt#b375]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:398:90)
-call core::result::unwrap_failed#5a05
-mov64 r2, 0
-ldxdw r4, [r10-1160]
-jle r4, r1, +0x590
-mov64 r3, 0
-ldxdw r4, [r10-1160]
-jeq r4, r1, +0x5a0
-mov64 r2, r3
-and64 r2, 1
-jne r2, 0, +0x5b0
-ldxdw r7, [r10-1176]
-stxw [r10-24], r7
-mov64 r1, r10
-add64 r1, -304
-mov64 r6, r10
-add64 r6, -24
-mov64 r2, r6
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#fdf6
-stdw [r10-56], 0
-stdw [r10-64], 1
-stdw [r10-72], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#91b5, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#a290]
-stxdw [r10-144], r1
-mov64 r1, r10
-add64 r1, -72
-stxdw [r10-152], r1
-lddw r1, 0xe0000020
-stxdw [r10-136], r1
-mov64 r2, r10
-add64 r2, -152
-mov64 r1, r6
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#0036
-jne r0, 0, +0x3188
-ldxdw r1, [r10-56]
-stxdw [r10-1064], r1
-ldxdw r1, [r10-64]
-stxdw [r10-1072], r1
-ldxdw r1, [r10-72]
-stxdw [r10-1080], r1
-ldxdw r1, [r10-304]
-stxdw [r10-1104], r1
-ldxdw r1, [r10-296]
-stxdw [r10-1096], r1
-ldxdw r1, [r10-288]
-stxdw [r10-1088], r1
-or64 r7, 6000
-stxw [r10-984], r7
-stb [r10-1056], 2
-stdw [r10-1136], 2
-mov64 r1, r10
-add64 r1, -1136
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-exit
-mov64 r2, r6
-ldxdw r7, [r10-1224]
-jeq r1, 0, +0x6f8
-mov64 r6, r2
-jlt r6, r0, +0x700
-mov64 r6, r0
-jle r7, r6, +0x708
-mov64 r1, r6
-mov64 r2, 0
-ldxdw r3, [r10-1168]
+jlt r1, r8, +0xe98
+mov64 r2, r8
+jne r1, 0, +0xea0
+jlt r8, r0, +0xa00
+mov64 r8, r0
+mov64 r1, r8
+mov64 r2, 0
+ldxdw r3, [r10-2536]
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
-sub64 r7, r6
+mov64 r4, 1
+jlt r9, r8, +0xa48
+mov64 r4, 0
+jlt r9, r8, +0x1cf0
+jeq r9, r8, +0x1c70
+sub64 r9, r8
+neg64 r4
 mov64 r1, r0
-mov64 r3, r7
-mov64 r4, 0
+mov64 r3, r9
 call __udivti3
-jne r1, 0, +0x3228
-mov64 r1, r6
-add64 r1, -1
-mov64 r7, 1
-mov64 r2, 1
-jgt r1, r6, +0x9d8
-mov64 r2, 0
-stxdw [r10-1240], r0
-mov64 r3, 0
-jne r2, 0, +0x9f8
-mov64 r3, r1
-stxdw [r10-1248], r3
-ldxdw r1, [r8+24]
+mov64 r6, r0
+jne r1, 0, +0x1c88
+ldxdw r1, [r7+24]
 call solana_account_info::AccountInfo::lamports#aa02
-mov64 r6, r0
-ldxdw r1, [r8+40]
-ldxdw r0, [r1+104]
-ldxdw r1, [r8+112]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-1112], r2
-ldxdw r2, [r1+16]
-stxdw [r10-1120], r2
-ldxdw r2, [r1+8]
-stxdw [r10-1128], r2
-ldxdw r1, [r1]
-stxdw [r10-1136], r1
-ldxdw r2, [r8+64]
-ldxdw r4, [r2+8]
-ldxdw r1, [r4]
-add64 r1, 1
-jeq r1, 0, +0xaa8
-mov64 r7, 0
-ldxdw r3, [r2]
-stxdw [r10-1256], r3
-stxdw [r4], r1
-jeq r7, 1, +0x3180
-ldxdw r5, [r2+16]
-ldxdw r1, [r5]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xaf8
-mov64 r3, 0
-stxdw [r10-1264], r4
-stxdw [r5], r1
-jeq r3, 1, +0x3180
-stxdw [r10-1272], r5
-ldxdw r1, [r8+80]
-ldxdw r3, [r1+8]
-stxdw [r10-1144], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xb58
-mov64 r4, 0
-ldxb r5, [r2+42]
-stxdw [r10-1320], r5
-ldxb r5, [r2+41]
-stxdw [r10-1312], r5
-ldxb r5, [r2+40]
-stxdw [r10-1304], r5
-ldxdw r5, [r2+32]
-stxdw [r10-1296], r5
-ldxdw r2, [r2+24]
-stxdw [r10-1288], r2
+stxdw [r10-2544], r0
+ldxdw r1, [r7+40]
+ldxdw r1, [r1+104]
+stxdw [r10-2536], r1
+ldxdw r2, [r7+64]
+mov64 r9, r10
+add64 r9, -2504
+mov64 r1, r9
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+80]
+mov64 r1, r10
+add64 r1, -2456
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r7+32]
+ldxdw r2, [r1+88]
+mov64 r1, r10
+add64 r1, -2408
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r7+72]
 ldxdw r2, [r1]
-stxdw [r10-1280], r2
-ldxdw r2, [r10-1144]
-stxdw [r2], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xc00
-mov64 r3, 0
-stxdw [r10-1328], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3180
-stxdw [r10-2128], r0
-ldxdw r2, [r8+32]
-ldxdw r2, [r2+88]
-ldxdw r5, [r2+8]
-ldxdw r3, [r5]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xc60
-mov64 r4, 0
-ldxb r0, [r1+42]
-stxdw [r10-1376], r0
-ldxb r0, [r1+41]
-stxdw [r10-1368], r0
-ldxb r0, [r1+40]
-stxdw [r10-1360], r0
-ldxdw r0, [r1+32]
-stxdw [r10-1352], r0
-ldxdw r1, [r1+24]
-stxdw [r10-1344], r1
-ldxdw r1, [r2]
-stxdw [r10-1336], r1
-stxdw [r5], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xd00
-mov64 r3, 0
-stxdw [r10-1384], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3180
-stxdw [r10-2136], r6
-ldxdw r1, [r8+72]
-ldxdw r1, [r1]
-ldxdw r9, [r1+8]
-ldxdw r3, [r9]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xd60
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-1408], r6
-ldxb r6, [r2+41]
-stxdw [r10-1400], r6
-ldxb r6, [r2+40]
-stxdw [r10-1392], r6
-ldxdw r6, [r2+32]
-stxdw [r10-1416], r6
-ldxdw r2, [r2+24]
-stxdw [r10-1424], r2
-ldxdw r7, [r1]
-stxdw [r9], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xdf8
-mov64 r3, 0
-stxdw [r10-1432], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3180
-ldxdw r2, [r8+88]
-ldxdw r2, [r2+32]
-ldxdw r3, [r2+8]
-stxdw [r10-1152], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xe58
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-1464], r6
-ldxb r6, [r1+41]
-stxdw [r10-1456], r6
-ldxb r6, [r1+40]
-stxdw [r10-1448], r6
-ldxdw r6, [r1+32]
-ldxdw r1, [r1+24]
-stxdw [r10-1472], r1
-ldxdw r1, [r2]
-stxdw [r10-1440], r1
-ldxdw r1, [r10-1152]
-stxdw [r1], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xef8
-mov64 r3, 0
-stxdw [r10-1488], r6
-stxdw [r10-1480], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3180
-stxdw [r10-1496], r7
-ldxdw r1, [r8+40]
-ldxdw r1, [r1+32]
-ldxdw r6, [r1+8]
-ldxdw r3, [r6]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xf60
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-1512], r7
-ldxb r7, [r2+41]
-stxdw [r10-1528], r7
-ldxb r7, [r2+40]
-stxdw [r10-1536], r7
-ldxdw r7, [r2+32]
-stxdw [r10-1520], r7
-ldxdw r7, [r2+24]
-ldxdw r2, [r1]
-stxdw [r10-1504], r2
-stxdw [r6], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xff8
-mov64 r3, 0
-stxdw [r10-1544], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3180
-stxdw [r10-1552], r7
-ldxdw r2, [r8+24]
-ldxdw r3, [r2+8]
-stxdw [r10-1160], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1058
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-1568], r7
-ldxb r7, [r1+41]
-stxdw [r10-1584], r7
-ldxb r7, [r1+40]
-stxdw [r10-1592], r7
-ldxdw r7, [r1+32]
-stxdw [r10-1576], r7
-ldxdw r7, [r1+24]
-ldxdw r1, [r2]
-stxdw [r10-1560], r1
-ldxdw r1, [r10-1160]
-stxdw [r1], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x10f8
-mov64 r3, 0
-stxdw [r10-1600], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3180
-stxdw [r10-1608], r7
-ldxdw r1, [r8+56]
-ldxdw r3, [r1+8]
-stxdw [r10-1168], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1158
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-1656], r7
-ldxb r7, [r2+41]
-stxdw [r10-1648], r7
-ldxb r7, [r2+40]
-stxdw [r10-1640], r7
-ldxdw r7, [r2+32]
-stxdw [r10-1632], r7
-ldxdw r2, [r2+24]
-stxdw [r10-1624], r2
-ldxdw r2, [r1]
-stxdw [r10-1616], r2
-ldxdw r2, [r10-1168]
-stxdw [r2], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1200
-mov64 r3, 0
-stxdw [r10-1664], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3180
-ldxdw r2, [r8+48]
-ldxdw r3, [r2+8]
-stxdw [r10-1176], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1258
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-1704], r7
-ldxb r7, [r1+41]
-stxdw [r10-1696], r7
-ldxb r7, [r1+40]
-stxdw [r10-1688], r7
-ldxdw r7, [r1+32]
-stxdw [r10-1680], r7
-ldxdw r1, [r1+24]
-stxdw [r10-1672], r1
-ldxdw r7, [r2]
-ldxdw r1, [r10-1176]
-stxdw [r1], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x12f8
-mov64 r3, 0
-stxdw [r10-1712], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3180
-stxdw [r10-1720], r7
-ldxdw r1, [r8+96]
-ldxdw r3, [r1+8]
-stxdw [r10-1184], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1358
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-1760], r7
-ldxb r7, [r2+41]
-stxdw [r10-1752], r7
-ldxb r7, [r2+40]
-stxdw [r10-1744], r7
-ldxdw r7, [r2+32]
-stxdw [r10-1736], r7
-ldxdw r2, [r2+24]
-stxdw [r10-1728], r2
-ldxdw r7, [r1]
-ldxdw r2, [r10-1184]
-stxdw [r2], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x13f8
-mov64 r3, 0
-mov64 r0, r6
-stxdw [r10-1768], r7
-stxdw [r10-1776], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3180
-mov64 r6, r5
-ldxdw r2, [r8+104]
-ldxdw r7, [r2+8]
-ldxdw r3, [r7]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1460
-mov64 r4, 0
-ldxb r5, [r1+42]
-stxdw [r10-1824], r5
-ldxb r5, [r1+41]
-stxdw [r10-1816], r5
-ldxb r5, [r1+40]
-stxdw [r10-1808], r5
-ldxdw r5, [r1+32]
-stxdw [r10-1800], r5
-ldxdw r1, [r1+24]
+mov64 r1, r10
+add64 r1, -2360
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r7+88]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -2312
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r7+40]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -2264
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+24]
+mov64 r1, r10
+add64 r1, -2216
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+56]
+mov64 r1, r10
+add64 r1, -2168
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+96]
+mov64 r1, r10
+add64 r1, -2120
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+48]
+mov64 r1, r10
+add64 r1, -2072
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+104]
+mov64 r1, r10
+add64 r1, -2024
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+112]
+mov64 r1, r10
+add64 r1, -1976
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+120]
+mov64 r1, r10
+add64 r1, -1928
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r7+128]
+mov64 r1, r10
+add64 r1, -1880
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r10-2528]
+ldxb r1, [r1+34]
+stxb [r10-1825], r1
+mov64 r1, r10
+add64 r1, -1825
+stxdw [r10-1808], r1
+lddw r1, .rodata[736f6c2d7661756c]
+stxdw [r10-1824], r1
+stdw [r10-1800], 1
+stdw [r10-1816], 9
+mov64 r1, r10
+add64 r1, -1824
 stxdw [r10-1792], r1
-ldxdw r1, [r2]
-stxdw [r10-1784], r1
-stxdw [r7], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-mov64 r5, r6
-jeq r1, 0, +0x1508
-mov64 r3, 0
-stxdw [r10-1832], r4
-stxdw [r4], r1
-mov64 r6, r0
-jeq r3, 1, +0x3180
-ldxdw r1, [r8+112]
-ldxdw r3, [r1+8]
-stxdw [r10-1208], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-mov64 r0, r9
-jeq r3, 0, +0x1570
-mov64 r4, 0
-ldxb r9, [r2+42]
-stxdw [r10-1880], r9
-ldxb r9, [r2+41]
-stxdw [r10-1872], r9
-ldxb r9, [r2+40]
-stxdw [r10-1864], r9
-ldxdw r9, [r2+32]
-stxdw [r10-1856], r9
-ldxdw r2, [r2+24]
-stxdw [r10-1848], r2
-ldxdw r2, [r1]
-stxdw [r10-1840], r2
-ldxdw r2, [r10-1208]
-stxdw [r2], r3
-ldxdw r9, [r10-1192]
-jeq r4, 1, +0x3180
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1620
-mov64 r3, 0
-stxdw [r10-1888], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3180
-stxdw [r10-1896], r6
-ldxdw r2, [r8+120]
-ldxdw r3, [r2+8]
-stxdw [r10-1224], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1680
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-1944], r6
-ldxb r6, [r1+41]
-stxdw [r10-1936], r6
-ldxb r6, [r1+40]
-stxdw [r10-1928], r6
-ldxdw r6, [r1+32]
-stxdw [r10-1920], r6
-ldxdw r1, [r1+24]
-stxdw [r10-1912], r1
-ldxdw r1, [r2]
-stxdw [r10-1904], r1
-ldxdw r1, [r10-1224]
-stxdw [r1], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1728
-mov64 r3, 0
-stxdw [r10-1952], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3180
-ldxdw r1, [r8+128]
-ldxdw r3, [r1+8]
-stxdw [r10-1232], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1780
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-1992], r6
-ldxb r6, [r2+41]
-stxdw [r10-1984], r6
-ldxb r6, [r2+40]
-stxdw [r10-1976], r6
-ldxdw r6, [r2+32]
-stxdw [r10-1968], r6
-ldxdw r2, [r2+24]
-stxdw [r10-1960], r2
-ldxdw r6, [r1]
-ldxdw r2, [r10-1232]
-stxdw [r2], r3
-jeq r4, 1, +0x3180
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1820
-mov64 r3, 0
-stxdw [r10-2008], r0
-stxdw [r10-2000], r5
-stxdw [r10-2016], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3180
-ldxb r2, [r1+42]
-stxdw [r10-2056], r2
-ldxb r2, [r1+41]
-stxdw [r10-2048], r2
-ldxb r2, [r1+40]
-stxdw [r10-2040], r2
-ldxdw r2, [r1+32]
-stxdw [r10-2032], r2
-ldxdw r1, [r1+24]
-stxdw [r10-2024], r1
-ldxb r1, [r9+34]
-mov64 r2, r10
-add64 r2, -305
-stxdw [r10-136], r2
-lddw r2, .rodata[736f6c2d7661756c]
-stxdw [r10-152], r2
-stxb [r10-305], r1
-mov64 r1, r10
-add64 r1, -152
-stxdw [r10-328], r1
-ldxdw r1, [r10-1136]
-stxdw [r10-72], r1
-ldxdw r1, [r10-1128]
-stxdw [r10-64], r1
-ldxdw r1, [r10-1120]
-stxdw [r10-56], r1
-ldxdw r1, [r10-1112]
-stxdw [r10-48], r1
-stdw [r10-128], 1
-stdw [r10-144], 9
-stdw [r10-320], 2
+stdw [r10-1784], 2
+mov64 r1, r10
+add64 r1, -1008
+mov64 r2, r9
+mov64 r3, 672
+call memcpy
+lddw r1, 0xcf5a6693f6e05601
+stxdw [r10-320], r1
+lddw r1, 0xaa5b17bf6815db44
+stxdw [r10-312], r1
+lddw r1, 0x3bffd2f597cb8951
+stxdw [r10-304], r1
+lddw r1, 0xb0186dfdb62b5d65
+stxdw [r10-296], r1
+mov64 r1, r10
+add64 r1, -1792
+stxdw [r10-336], r1
+stdw [r10-1016], 0
+stdw [r10-1024], 8
+stdw [r10-1032], 0
+stdw [r10-328], 1
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
-ldxdw r3, [r1]
-mov64 r2, 0
+ldxdw r2, [r1]
+mov64 r1, 0
 mov64 r4, 1
-mov64 r1, r3
-add64 r1, -96
-jle r1, r3, +0x25e8
-jeq r4, 0, +0x25f8
-lddw r1, 0x300007fa0
-jne r3, 0, +0x2618
-lddw r2, 0x300000008
-jlt r1, r2, +0x2640
-lddw r2, 0x300000000
-stxdw [r2], r1
-ldxdw r5, [r8+152]
-ldxdw r2, [r5+8]
-ldxdw r3, [r2]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1a18
-mov64 r4, 0
-ldxdw r9, [r5]
-stxdw [r2], r3
-and64 r4, 1
-jne r4, 0, +0x3180
-ldxdw r4, [r5+16]
-ldxdw r3, [r4]
-add64 r3, 1
-mov64 r0, 1
-jeq r3, 0, +0x1a68
-mov64 r0, 0
-stxdw [r10-2064], r6
-stxdw [r10-1192], r7
-stxdw [r4], r3
-jeq r0, 1, +0x3180
-ldxdw r6, [r8+160]
-ldxdw r0, [r6+8]
-ldxdw r7, [r0]
-add64 r7, 1
-mov64 r8, 1
-jeq r7, 0, +0x1ac0
-mov64 r8, 0
-ldxb r3, [r5+42]
-stxdw [r10-2096], r3
-ldxb r3, [r5+41]
-stxdw [r10-2088], r3
-ldxb r3, [r5+40]
-stxdw [r10-2080], r3
-ldxdw r3, [r5+32]
-stxdw [r10-2072], r3
-ldxdw r3, [r5+24]
-ldxdw r5, [r6]
-stxdw [r10-2104], r5
-stxdw [r0], r7
-jeq r8, 1, +0x3180
-stxdw [r10-2112], r2
-ldxdw r5, [r6+16]
-ldxdw r7, [r5]
-add64 r7, 1
-mov64 r8, 1
-jeq r7, 0, +0x1b60
-mov64 r8, 0
-mov64 r2, r3
-stxdw [r10-2120], r9
-stxdw [r5], r7
-jeq r8, 1, +0x3180
-ldxdw r7, [r6+24]
-ldxdw r8, [r6+32]
-ldxb r9, [r6+40]
-ldxb r3, [r6+41]
-ldxb r6, [r6+42]
-stxb [r1+90], r6
-stxb [r1+89], r3
-stxb [r1+88], r9
-stxdw [r1+80], r8
-stxdw [r1+72], r7
-stxdw [r1+64], r5
-stxdw [r1+56], r0
-ldxdw r3, [r10-2104]
-stxdw [r1+48], r3
-ldxdw r3, [r10-2096]
-stxb [r1+42], r3
-ldxdw r3, [r10-2088]
-stxb [r1+41], r3
-ldxdw r3, [r10-2080]
-stxb [r1+40], r3
-ldxdw r3, [r10-2072]
-stxdw [r1+32], r3
-stxdw [r1+24], r2
-stxdw [r1+16], r4
-ldxdw r3, [r10-2112]
-stxdw [r1+8], r3
-ldxdw r2, [r10-2120]
-stxdw [r1], r2
-ldxdw r2, [r10-1320]
-stxb [r10-1070], r2
-ldxdw r2, [r10-1312]
-stxb [r10-1071], r2
-ldxdw r2, [r10-1304]
-stxb [r10-1072], r2
-ldxdw r2, [r10-1296]
-stxdw [r10-1080], r2
-ldxdw r2, [r10-1288]
-stxdw [r10-1088], r2
-ldxdw r2, [r10-1272]
-stxdw [r10-1096], r2
-ldxdw r2, [r10-1264]
-stxdw [r10-1104], r2
-ldxdw r2, [r10-1256]
-stxdw [r10-1112], r2
-stxdw [r10-1128], r1
-ldxdw r1, [r10-1376]
-stxb [r10-1022], r1
-ldxdw r1, [r10-1368]
-stxb [r10-1023], r1
-ldxdw r1, [r10-1360]
-stxb [r10-1024], r1
-ldxdw r1, [r10-1352]
-stxdw [r10-1032], r1
-ldxdw r1, [r10-1344]
-stxdw [r10-1040], r1
-ldxdw r1, [r10-1328]
-stxdw [r10-1048], r1
-ldxdw r1, [r10-1144]
-stxdw [r10-1056], r1
-ldxdw r1, [r10-1280]
-stxdw [r10-1064], r1
-ldxw r1, [r10-304]
-stxw [r10-1069], r1
-ldxb r1, [r10-300]
-stxb [r10-1065], r1
-stdw [r10-1120], 2
-stdw [r10-1136], 2
-ldxdw r1, [r10-1336]
-stxdw [r10-1016], r1
-ldxdw r1, [r10-2000]
-stxdw [r10-1008], r1
-ldxdw r1, [r10-1384]
-stxdw [r10-1000], r1
-ldxdw r1, [r10-1424]
-stxdw [r10-992], r1
-ldxdw r1, [r10-1416]
-stxdw [r10-984], r1
-ldxdw r1, [r10-1408]
-stxb [r10-974], r1
-ldxdw r1, [r10-1400]
-stxb [r10-975], r1
-ldxdw r1, [r10-1392]
-stxb [r10-976], r1
-ldxdw r1, [r10-1496]
-stxdw [r10-968], r1
-ldxdw r1, [r10-2008]
-stxdw [r10-960], r1
-ldxdw r1, [r10-1432]
-stxdw [r10-952], r1
-ldxdw r1, [r10-1472]
-stxdw [r10-944], r1
-ldxdw r1, [r10-1488]
-stxdw [r10-936], r1
-ldxdw r1, [r10-1464]
-stxb [r10-926], r1
-ldxdw r1, [r10-1456]
-stxb [r10-927], r1
-ldxdw r1, [r10-1448]
-stxb [r10-928], r1
-ldxb r1, [r10-20]
-stxb [r10-1017], r1
-ldxw r1, [r10-24]
-stxw [r10-1021], r1
-ldxw r1, [r10-333]
-stxw [r10-973], r1
-ldxb r1, [r10-329]
-stxb [r10-969], r1
-ldxdw r1, [r10-1536]
-stxb [r10-880], r1
-ldxdw r1, [r10-1528]
-stxb [r10-879], r1
-ldxdw r1, [r10-1512]
-stxb [r10-878], r1
-ldxdw r1, [r10-1520]
-stxdw [r10-888], r1
-ldxdw r1, [r10-1552]
-stxdw [r10-896], r1
-ldxdw r1, [r10-1480]
-stxdw [r10-904], r1
-ldxdw r1, [r10-1152]
-stxdw [r10-912], r1
-ldxdw r1, [r10-1440]
-stxdw [r10-920], r1
-ldxdw r1, [r10-1592]
-stxb [r10-832], r1
-ldxdw r1, [r10-1584]
-stxb [r10-831], r1
-ldxdw r1, [r10-1568]
-stxb [r10-830], r1
-ldxdw r1, [r10-1576]
-stxdw [r10-840], r1
-ldxdw r1, [r10-1608]
-stxdw [r10-848], r1
-ldxdw r1, [r10-1544]
-stxdw [r10-856], r1
-ldxdw r1, [r10-1896]
-stxdw [r10-864], r1
-ldxdw r1, [r10-1504]
-stxdw [r10-872], r1
-ldxb r1, [r10-334]
-stxb [r10-921], r1
-ldxw r1, [r10-338]
-stxw [r10-925], r1
-ldxw r1, [r10-343]
-stxw [r10-877], r1
-ldxb r1, [r10-339]
-stxb [r10-873], r1
-ldxb r1, [r10-344]
-stxb [r10-825], r1
-ldxw r1, [r10-348]
-stxw [r10-829], r1
-ldxdw r1, [r10-1656]
-stxb [r10-782], r1
-ldxdw r1, [r10-1648]
-stxb [r10-783], r1
-ldxdw r1, [r10-1640]
-stxb [r10-784], r1
-ldxdw r1, [r10-1632]
-stxdw [r10-792], r1
-ldxdw r1, [r10-1624]
-stxdw [r10-800], r1
-ldxdw r1, [r10-1600]
-stxdw [r10-808], r1
-ldxdw r1, [r10-1160]
-stxdw [r10-816], r1
-ldxdw r1, [r10-1560]
-stxdw [r10-824], r1
-ldxb r1, [r10-349]
-stxb [r10-777], r1
-ldxw r1, [r10-353]
-stxw [r10-781], r1
-ldxdw r1, [r10-1704]
-stxb [r10-734], r1
-ldxdw r1, [r10-1696]
-stxb [r10-735], r1
-ldxdw r1, [r10-1688]
-stxb [r10-736], r1
-ldxdw r1, [r10-1680]
-stxdw [r10-744], r1
-ldxdw r1, [r10-1672]
-stxdw [r10-752], r1
-ldxdw r1, [r10-1664]
-stxdw [r10-760], r1
-ldxdw r1, [r10-1168]
-stxdw [r10-768], r1
-ldxdw r1, [r10-1616]
-stxdw [r10-776], r1
-ldxb r1, [r10-354]
-stxb [r10-729], r1
-ldxw r1, [r10-358]
-stxw [r10-733], r1
-ldxdw r1, [r10-1824]
-stxb [r10-686], r1
-ldxdw r1, [r10-1816]
-stxb [r10-687], r1
-ldxdw r1, [r10-1808]
-stxb [r10-688], r1
-ldxdw r1, [r10-1800]
-stxdw [r10-696], r1
-ldxdw r1, [r10-1792]
-stxdw [r10-704], r1
-ldxdw r1, [r10-1776]
-stxdw [r10-712], r1
-ldxdw r1, [r10-1184]
-stxdw [r10-720], r1
-ldxdw r1, [r10-1768]
-stxdw [r10-728], r1
-ldxb r1, [r10-359]
-stxb [r10-681], r1
-ldxw r1, [r10-363]
-stxw [r10-685], r1
-ldxdw r1, [r10-1760]
-stxb [r10-638], r1
-ldxdw r1, [r10-1752]
-stxb [r10-639], r1
-ldxdw r1, [r10-1744]
-stxb [r10-640], r1
-ldxdw r1, [r10-1736]
-stxdw [r10-648], r1
-ldxdw r1, [r10-1728]
-stxdw [r10-656], r1
-ldxdw r1, [r10-1712]
-stxdw [r10-664], r1
-ldxdw r1, [r10-1176]
-stxdw [r10-672], r1
-ldxdw r1, [r10-1720]
-stxdw [r10-680], r1
-ldxb r1, [r10-364]
-stxb [r10-633], r1
-ldxw r1, [r10-368]
-stxw [r10-637], r1
-ldxdw r1, [r10-1880]
-stxb [r10-590], r1
-ldxdw r1, [r10-1872]
-stxb [r10-591], r1
-ldxdw r1, [r10-1864]
-stxb [r10-592], r1
-ldxdw r1, [r10-1856]
-stxdw [r10-600], r1
-ldxdw r1, [r10-1848]
-stxdw [r10-608], r1
-ldxdw r1, [r10-1832]
-stxdw [r10-616], r1
-ldxdw r1, [r10-1192]
-stxdw [r10-624], r1
-ldxdw r1, [r10-1784]
-stxdw [r10-632], r1
-ldxb r1, [r10-369]
-stxb [r10-585], r1
-ldxw r1, [r10-373]
-stxw [r10-589], r1
-ldxdw r1, [r10-1944]
-stxb [r10-542], r1
-ldxdw r1, [r10-1936]
-stxb [r10-543], r1
-ldxdw r1, [r10-1928]
-stxb [r10-544], r1
-ldxdw r1, [r10-1920]
-stxdw [r10-552], r1
-ldxdw r1, [r10-1912]
-stxdw [r10-560], r1
-ldxdw r1, [r10-1888]
-stxdw [r10-568], r1
-ldxdw r1, [r10-1208]
-stxdw [r10-576], r1
-ldxdw r1, [r10-1840]
-stxdw [r10-584], r1
-ldxb r1, [r10-374]
-stxb [r10-537], r1
-ldxw r1, [r10-378]
-stxw [r10-541], r1
-ldxdw r1, [r10-1992]
-stxb [r10-494], r1
-ldxdw r1, [r10-1984]
-stxb [r10-495], r1
-ldxdw r1, [r10-1976]
-stxb [r10-496], r1
-ldxdw r1, [r10-1968]
-stxdw [r10-504], r1
-ldxdw r1, [r10-1960]
-stxdw [r10-512], r1
-ldxdw r1, [r10-1952]
-stxdw [r10-520], r1
-ldxdw r1, [r10-1224]
-stxdw [r10-528], r1
-ldxdw r1, [r10-1904]
-stxdw [r10-536], r1
-ldxb r1, [r10-379]
-stxb [r10-489], r1
-ldxw r1, [r10-383]
-stxw [r10-493], r1
-ldxdw r1, [r10-2056]
-stxb [r10-446], r1
-ldxdw r1, [r10-2048]
-stxb [r10-447], r1
-ldxdw r1, [r10-2040]
-stxb [r10-448], r1
-ldxdw r1, [r10-2032]
-stxdw [r10-456], r1
-ldxdw r1, [r10-2024]
-stxdw [r10-464], r1
-ldxdw r1, [r10-2016]
-stxdw [r10-472], r1
-ldxdw r1, [r10-1232]
-stxdw [r10-480], r1
-ldxdw r1, [r10-2064]
-stxdw [r10-488], r1
-ldxb r1, [r10-384]
-stxb [r10-441], r1
-ldxw r1, [r10-388]
-stxw [r10-445], r1
-mov64 r1, r10
-add64 r1, -328
-stxdw [r10-440], r1
-stdw [r10-432], 1
-ldxdw r1, [r10-48]
-stxdw [r10-400], r1
-ldxdw r1, [r10-56]
-stxdw [r10-408], r1
-ldxdw r1, [r10-64]
-stxdw [r10-416], r1
-ldxdw r1, [r10-72]
-stxdw [r10-424], r1
-mov64 r1, r10
-add64 r1, -1136
-ldxdw r2, [r10-1240]
-ldxdw r3, [r10-1248]
-call mayhem_program::pump::cpi::sell#a6e2
-mov64 r7, r0
-mov64 r6, r1
-jne r7, 2, +0x2658
-ldxdw r8, [r10-1200]
-ja +0x2800
-mov64 r4, 0
-jne r4, 0, +0x1998
-mov64 r2, r1
-lddw r1, 0x300007fa0
-jeq r3, 0, +0x19b0
-and64 r2, -8
-mov64 r1, r2
-lddw r2, 0x300000008
-jge r1, r2, +0x19c8
+mov64 r3, r2
+add64 r3, -96
+jgt r3, r2, +0xeb8
+mov64 r4, 0
+jeq r4, 0, +0xec0
+lddw r9, 0x300007fa0
+jeq r2, 0, +0xee0
+and64 r1, -8
+mov64 r9, r1
+stxdw [r10-2512], r6
+lddw r1, 0x300000008
+jge r9, r1, +0xf00
 mov64 r1, 8
 mov64 r2, 96
 call alloc::alloc::handle_alloc_error#decd
-mov64 r1, r10
-add64 r1, -24
-lddw r2, .rodata[08000000]
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#fdf6
-stdw [r10-288], 0
-stdw [r10-296], 1
-stdw [r10-304], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#91b5, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#a290]
-stxdw [r10-64], r1
-mov64 r1, r10
-add64 r1, -304
-stxdw [r10-72], r1
-lddw r1, 0xe0000020
-stxdw [r10-56], r1
-mov64 r2, r10
-add64 r2, -72
-lddw r1, .rodata[08000000]
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#0036
-ldxdw r8, [r10-1200]
-jne r0, 0, +0x3188
-ldxdw r1, [r10-288]
-stxdw [r10-1064], r1
-ldxdw r1, [r10-296]
-stxdw [r10-1072], r1
-ldxdw r1, [r10-304]
-stxdw [r10-1080], r1
-ldxdw r1, [r10-24]
-stxdw [r10-1104], r1
-ldxdw r1, [r10-16]
-stxdw [r10-1096], r1
-ldxdw r1, [r10-8]
-stxdw [r10-1088], r1
-lddw r1, .rodata[70726f6772616d73]
-stxdw [r10-1128], r1
-stw [r10-984], 6008
-stb [r10-1056], 2
-stw [r10-1112], 436
-stdw [r10-1120], 58
-stdw [r10-1136], 0
-mov64 r1, r10
-add64 r1, -1136
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-jeq r7, 0, +0x27f0
-ldxdw r2, [r6]
-ldxb r2, [r6+40]
-ja +0x27f8
-ldxdw r2, [r6]
-jne r0, 2, +0x908
-ldxdw r1, [r8+40]
+jeq r1, 0, +0x9f0
+mov64 r8, r2
+jge r8, r0, +0x9f8
+ja +0xa00
+jne r4, 0, +0xe38
+mov64 r1, r3
+lddw r9, 0x300007fa0
+jne r2, 0, +0xe50
+stxdw [r10-2512], r6
+lddw r1, 0x300000008
+jlt r9, r1, +0xe80
+lddw r1, 0x300000000
+stxdw [r1], r9
+ldxdw r6, [r10-2552]
+ldxdw r2, [r6+152]
+mov64 r7, r10
+add64 r7, -56
+mov64 r1, r7
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+160]
+mov64 r6, r10
+add64 r6, -136
+mov64 r1, r6
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r1, r9
+mov64 r2, r7
+mov64 r3, 48
+call memcpy
+mov64 r1, r9
+add64 r1, 48
+mov64 r2, r6
+mov64 r3, 48
+call memcpy
+mov64 r6, r10
+add64 r6, -1032
+mov64 r1, r6
+call core::ptr::drop_in_place<alloc::vec::Vec<solana_account_info::AccountInfo>>#d146
+stxdw [r10-1024], r9
+stdw [r10-1016], 2
+stdw [r10-1032], 2
+mov64 r9, r10
+add64 r9, -1776
+mov64 r1, r9
+mov64 r2, r6
+mov64 r3, 744
+call memcpy
+mov64 r1, r8
+add64 r1, -1
+mov64 r3, 0
+mov64 r4, 1
+jgt r1, r8, +0x1080
+mov64 r4, 0
+ldxdw r2, [r10-2512]
+jeq r4, 0, +0x1090
+mov64 r1, r9
+call mayhem_program::pump::cpi::sell#a6e2
+jne r0, 2, +0x430
+ja +0x10b0
+ldxdw r2, [r10-2512]
+jne r4, 0, +0x1060
+mov64 r3, r1
+mov64 r1, r9
+call mayhem_program::pump::cpi::sell#a6e2
+jne r0, 2, +0x430
+ldxdw r7, [r10-2552]
+ldxdw r1, [r7+40]
 call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#d655
-jne r0, 2, +0x908
-ldxdw r1, [r8+72]
+jne r0, 2, +0x430
+ldxdw r1, [r7+72]
 call anchor_lang::accounts::account::Account<T>::reload#fa41
-jne r0, 2, +0x908
-ldxdw r1, [r8+24]
+jne r0, 2, +0x430
+ldxdw r1, [r7+24]
 call solana_account_info::AccountInfo::lamports#aa02
 mov64 r6, r0
-ldxdw r2, [r10-2136]
-jlt r6, r2, +0x3290
-ldxdw r1, [r8+40]
+ldxdw r2, [r10-2544]
+jlt r6, r2, +0x1d08
+ldxdw r1, [r7+40]
 ldxdw r1, [r1+104]
-ldxdw r4, [r10-2128]
-jlt r4, r1, +0x32a8
+ldxdw r8, [r10-2536]
+jlt r8, r1, +0x1d20
 sub64 r6, r2
-sub64 r4, r1
-ldxdw r2, [r8+16]
-ldxdw r1, [r8+8]
-stdw [r10-1112], 0
-stdw [r10-1120], 0
-stdw [r10-1128], 0
-stdw [r10-1136], 0
-add64 r1, 8
-add64 r2, 8
-mov64 r5, r10
-add64 r5, -1136
-mov64 r3, r6
-mov64 r7, r4
-call orig_fn_2ce68
-jne r0, 2, +0x908
-ldxdw r1, [r8+24]
+ldxdw r2, [r7+8]
+ldxdw r3, [r2+72]
+mov64 r0, 1
+jlt r3, r6, +0x1160
+mov64 r0, 0
+ldxdw r5, [r2+80]
+mov64 r4, r5
+sub64 r4, r0
+mov64 r0, r5
+xor64 r0, r4
+and64 r5, r0
+jslt r5, 0, +0x1460
+sub64 r3, r6
+stxdw [r2+72], r3
+stxdw [r2+80], r4
+ldxdw r2, [r7+16]
+ldxdw r3, [r2+40]
+mov64 r0, 1
+jlt r3, r6, +0x11d8
+mov64 r0, 0
+ldxdw r5, [r2+48]
+mov64 r4, r5
+sub64 r4, r0
+mov64 r0, r5
+xor64 r0, r4
+and64 r5, r0
+jsle r5, -1, +0x1460
+sub64 r8, r1
+sub64 r3, r6
+stxdw [r2+40], r3
+stxdw [r2+48], r4
+ldxdw r1, [r7+16]
+ldxdw r3, [r1+56]
+mov64 r2, r3
+add64 r2, r8
+mov64 r5, 1
+jlt r2, r3, +0x1268
+mov64 r5, 0
+ldxdw r4, [r1+64]
+mov64 r3, r4
+add64 r3, r5
+mov64 r5, r4
+xor64 r5, r3
+xor64 r4, -1
+and64 r4, r5
+jsle r4, -1, +0x1460
+stxdw [r10-2536], r8
+stxdw [r1+56], r2
+stxdw [r1+64], r3
+ldxdw r1, [r7+24]
 ldxdw r2, [r1]
 ldxdw r3, [r2+24]
+stxdw [r10-264], r3
+ldxdw r3, [r2+16]
+stxdw [r10-272], r3
+ldxdw r3, [r2+8]
 stxdw [r10-280], r3
-ldxdw r3, [r2+16]
-stxdw [r10-288], r3
-ldxdw r3, [r2+8]
-stxdw [r10-296], r3
 ldxdw r2, [r2]
-stxdw [r10-304], r2
-ldxdw r2, [r8+32]
+stxdw [r10-288], r2
+ldxdw r2, [r7+32]
 ldxdw r2, [r2+88]
 ldxdw r2, [r2]
 ldxdw r3, [r2+24]
+stxdw [r10-32], r3
+ldxdw r3, [r2+16]
+stxdw [r10-40], r3
+ldxdw r3, [r2+8]
 stxdw [r10-48], r3
-ldxdw r3, [r2+16]
-stxdw [r10-56], r3
-ldxdw r3, [r2+8]
-stxdw [r10-64], r3
 ldxdw r2, [r2]
-stxdw [r10-72], r2
+stxdw [r10-56], r2
 call solana_account_info::AccountInfo::lamports#aa02
-stxdw [r10-1144], r0
-ldxdw r1, [r8+40]
-ldxdw r1, [r1+104]
-stxdw [r10-1152], r1
-ldxdw r1, [r8+16]
+stxdw [r10-2568], r0
+ldxdw r1, [r7+16]
 ldxdw r2, [r1+48]
-stxdw [r10-1192], r2
+stxdw [r10-2528], r2
 ldxdw r2, [r1+40]
-stxdw [r10-1184], r2
+stxdw [r10-2512], r2
 ldxdw r2, [r1+64]
-stxdw [r10-1200], r2
-ldxdw r9, [r1+56]
+stxdw [r10-2584], r2
+ldxdw r1, [r1+56]
+stxdw [r10-2576], r1
+ldxdw r1, [r7+8]
+ldxdw r2, [r1+80]
+stxdw [r10-2560], r2
+ldxdw r1, [r1+72]
+stxdw [r10-2544], r1
+ldxdw r1, [r7+40]
+ldxdw r9, [r1+104]
+ldxdw r1, [r7+72]
+ldxdw r2, [r1+80]
+stxdw [r10-2592], r2
+ldxdw r8, [r1+72]
+mov64 r1, r10
+add64 r1, -136
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-136]
+jne r1, 1, +0x1470
+ldxw r2, [r10-128]
+ldxw r1, [r10-132]
+call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
+ja +0x430
+call mayhem_program::instructions::pump_interface::buy::math#52f7
+ja +0x430
+ldxdw r1, [r10-96]
+stxdw [r10-2600], r1
+ldxdw r1, [r7+16]
 ldxdw r1, [r1+80]
-stxdw [r10-1168], r1
-ldxdw r1, [r8+72]
-ldxdw r2, [r1+80]
-stxdw [r10-1160], r2
-ldxdw r1, [r1+72]
-stxdw [r10-1176], r1
-mov64 r1, r10
-add64 r1, -152
-call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-152]
-jne r1, 1, +0x2a88
-ldxw r2, [r10-144]
-ldxw r1, [r10-148]
-call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-ja +0x908
-ldxdw r1, [r10-112]
-stxdw [r10-1208], r1
-ldxdw r1, [r8+72]
+stxdw [r10-2608], r1
+ldxdw r1, [r7+72]
 ldxdw r2, [r1+96]
-stxdw [r10-1240], r2
+stxdw [r10-2616], r2
 ldxdw r1, [r1+88]
-stxdw [r10-1248], r1
-ldxdw r1, [r8+8]
-ldxw r2, [r1+248]
-stxdw [r10-1256], r2
+stxdw [r10-2624], r1
+ldxdw r1, [r7+8]
+ldxw r1, [r1+224]
+stxdw [r10-2648], r1
+ldxdw r1, [r7+24]
+call solana_account_info::AccountInfo::lamports#aa02
+ldxdw r1, [r7+8]
 ldxdw r2, [r1+72]
-stxdw [r10-1264], r2
+stxdw [r10-2632], r2
 ldxdw r1, [r1+80]
-stxdw [r10-1272], r1
-ldxdw r1, [r8+24]
-call solana_account_info::AccountInfo::lamports#aa02
-stxdw [r10-1224], r0
-ldxdw r1, [r8+72]
-ldxdw r2, [r1+80]
-stxdw [r10-1232], r2
-ldxdw r2, [r1+96]
-ldxdw r1, [r8+16]
-ldxdw r3, [r1+40]
-ldxdw r4, [r1+48]
-ldxdw r1, [r8+8]
-ldxdw r5, [r1+72]
-ldxdw r1, [r1+80]
-ldxdw r0, [r10-296]
-stxdw [r10-1128], r0
-ldxdw r0, [r10-288]
-stxdw [r10-1120], r0
-ldxdw r0, [r10-280]
-stxdw [r10-1112], r0
-ldxdw r0, [r10-304]
-stxdw [r10-1136], r0
-stdw [r10-856], 0
-stdw [r10-864], 0
-stdw [r10-872], 0
-stdw [r10-880], 0
-stb [r10-842], 1
-ldxdw r0, [r10-72]
-stxdw [r10-1104], r0
-ldxdw r0, [r10-64]
-stxdw [r10-1096], r0
-ldxdw r0, [r10-56]
-stxdw [r10-1088], r0
-ldxdw r0, [r10-48]
-stxdw [r10-1080], r0
-ldxdw r0, [r10-1272]
-stxdw [r10-984], r0
-ldxdw r0, [r10-1264]
-stxdw [r10-992], r0
-ldxdw r0, [r10-1200]
-stxdw [r10-1016], r0
-stxdw [r10-1024], r9
-ldxdw r0, [r10-1192]
-stxdw [r10-1032], r0
-ldxdw r0, [r10-1184]
-stxdw [r10-1040], r0
-ldxdw r0, [r10-1216]
-stxh [r10-844], r0
-ldxdw r0, [r10-1256]
-stxw [r10-848], r0
-ldxdw r0, [r10-1248]
-stxdw [r10-952], r0
-ldxdw r0, [r10-1240]
-stxdw [r10-960], r0
-ldxdw r0, [r10-1168]
-stxdw [r10-968], r0
-ldxdw r0, [r10-1208]
-stxdw [r10-976], r0
-ldxdw r0, [r10-1176]
-stxdw [r10-1000], r0
-ldxdw r0, [r10-1160]
-stxdw [r10-1008], r0
-ldxdw r0, [r10-1152]
-stxdw [r10-1048], r0
-ldxdw r0, [r10-1144]
-stxdw [r10-1056], r0
-stxdw [r10-1064], r7
-stxdw [r10-1072], r6
-stxdw [r10-896], r1
-stxdw [r10-904], r5
-stxdw [r10-920], r4
-stxdw [r10-928], r3
-stxdw [r10-888], r2
-ldxdw r1, [r10-1232]
-stxdw [r10-912], r1
-ldxdw r1, [r10-1224]
-stxdw [r10-936], r1
-stxdw [r10-944], r6
-mov64 r1, r10
-add64 r1, -152
+stxdw [r10-2640], r1
+ldxdw r1, [r7+72]
+ldxdw r3, [r1+96]
+ldxdw r4, [r1+80]
+ldxdw r1, [r7+16]
+ldxdw r5, [r1+40]
+ldxdw r1, [r1+48]
+ldxdw r2, [r10-280]
+stxdw [r10-1024], r2
+ldxdw r2, [r10-272]
+stxdw [r10-1016], r2
+ldxdw r2, [r10-264]
+stxdw [r10-1008], r2
+ldxdw r2, [r10-288]
+stxdw [r10-1032], r2
+stdw [r10-944], 0
+stdw [r10-952], 0
+stdw [r10-960], 0
+stdw [r10-968], 0
+stb [r10-738], 1
+ldxdw r2, [r10-56]
+stxdw [r10-1000], r2
+ldxdw r2, [r10-48]
+stxdw [r10-992], r2
+ldxdw r2, [r10-40]
+stxdw [r10-984], r2
+ldxdw r2, [r10-32]
+stxdw [r10-976], r2
+stxdw [r10-832], r9
+ldxdw r2, [r10-2568]
+stxdw [r10-840], r2
+ldxdw r2, [r10-2536]
+stxdw [r10-848], r2
+stxdw [r10-856], r6
+ldxdw r2, [r10-2584]
+stxdw [r10-912], r2
+ldxdw r2, [r10-2576]
+stxdw [r10-920], r2
+ldxdw r2, [r10-2528]
+stxdw [r10-928], r2
+ldxdw r2, [r10-2512]
+stxdw [r10-936], r2
+stxdw [r10-816], r8
+ldxdw r2, [r10-2592]
+stxdw [r10-824], r2
+ldxdw r2, [r10-2560]
+stxdw [r10-896], r2
+ldxdw r2, [r10-2544]
+stxdw [r10-904], r2
+ldxdw r2, [r10-2520]
+stxh [r10-740], r2
+ldxdw r2, [r10-2648]
+stxw [r10-744], r2
+ldxdw r2, [r10-2624]
+stxdw [r10-784], r2
+ldxdw r2, [r10-2616]
+stxdw [r10-792], r2
+ldxdw r2, [r10-2608]
+stxdw [r10-800], r2
+ldxdw r2, [r10-2600]
+stxdw [r10-808], r2
+stxdw [r10-768], r0
+stxdw [r10-776], r6
+stxdw [r10-880], r1
+stxdw [r10-888], r5
+stxdw [r10-760], r4
+ldxdw r1, [r10-2640]
+stxdw [r10-864], r1
+ldxdw r1, [r10-2632]
+stxdw [r10-872], r1
+stxdw [r10-752], r3
+mov64 r1, r10
+add64 r1, -136
 mov64 r6, r10
-add64 r6, -1136
+add64 r6, -1032
 mov64 r2, r6
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#c4f7
-ldxdw r1, [r10-136]
-stxdw [r10-64], r1
-ldxdw r1, [r10-144]
-stxdw [r10-72], r1
-mov64 r1, r10
-add64 r1, -72
-mov64 r9, 1
+ldxdw r1, [r10-120]
+stxdw [r10-48], r1
+ldxdw r1, [r10-128]
+stxdw [r10-56], r1
+mov64 r1, r10
+add64 r1, -56
+mov64 r8, 1
 mov64 r2, 1
 call anchor_lang::solana_program::log::sol_log_data#9398
-ldxdw r2, [r8+136]
-mov64 r1, r10
-add64 r1, -272
-call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
-mov64 r1, r10
-add64 r1, -224
+ldxdw r2, [r7+136]
+mov64 r1, r10
+add64 r1, -256
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r1, r10
+add64 r1, -208
 mov64 r2, r6
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#c4f7
 lddw r1, .rodata[e445a52e51cb9a1d]
 lddw r2, .rodata[e445a52e51cb9a1d]
 add64 r2, 8
-stxdw [r10-160], r2
+stxdw [r10-144], r2
+stxdw [r10-152], r1
+ldxdw r1, [r10-208]
 stxdw [r10-168], r1
-ldxdw r1, [r10-224]
+ldxdw r1, [r10-200]
+stxdw [r10-176], r1
 stxdw [r10-184], r1
-ldxdw r1, [r10-216]
-stxdw [r10-192], r1
-stxdw [r10-200], r1
-ldxdw r2, [r10-208]
+ldxdw r2, [r10-192]
 add64 r1, r2
-stxdw [r10-176], r1
-mov64 r1, r10
-add64 r1, -304
+stxdw [r10-160], r1
+mov64 r1, r10
+add64 r1, -288
 mov64 r2, r10
-add64 r2, -200
+add64 r2, -184
 call <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter#1a04
-ldxdw r6, [r10-296]
-ldxdw r7, [r10-288]
+ldxdw r6, [r10-280]
+ldxdw r7, [r10-272]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
 mov64 r3, 0
 mov64 r1, r2
 add64 r1, -34
-jle r1, r2, +0x30b8
-jeq r9, 0, +0x30c8
+jgt r1, r2, +0x1948
+mov64 r8, 0
+jeq r8, 0, +0x1950
 lddw r1, 0x300007fde
-jne r2, 0, +0x30e8
+jeq r2, 0, +0x1970
+mov64 r1, r3
 lddw r2, 0x300000008
-jlt r1, r2, +0x3108
+jge r1, r2, +0x1988
+mov64 r1, 1
+mov64 r2, 34
+call alloc::alloc::handle_alloc_error#decd
+jne r8, 0, +0x18f8
+mov64 r3, r1
+lddw r1, 0x300007fde
+jne r2, 0, +0x1910
+lddw r2, 0x300000008
+jlt r1, r2, +0x1930
 lddw r2, 0x300000000
 stxdw [r2], r1
-ldxdw r2, [r10-272]
+ldxdw r2, [r10-256]
 ldxdw r3, [r2]
 ldxdw r4, [r2+8]
 ldxdw r5, [r2+16]
@@ -1499,22 +799,22 @@
 stxdw [r1+8], r4
 stxdw [r1], r3
 sth [r1+32], 1
-stxdw [r10-64], r1
+stxdw [r10-48], r1
+stdw [r10-40], 1
 stdw [r10-56], 1
-stdw [r10-72], 1
 mov64 r8, r10
-add64 r8, -152
+add64 r8, -136
 mov64 r5, r10
-add64 r5, -72
+add64 r5, -56
 mov64 r1, r8
 lddw r2, .rodata[052ae5d7a7daa724]
 mov64 r3, r6
 mov64 r4, r7
 call solana_instruction::Instruction::new_with_bytes#33e2
 mov64 r6, r10
-add64 r6, -72
+add64 r6, -56
 mov64 r2, r10
-add64 r2, -272
+add64 r2, -256
 mov64 r1, r6
 mov64 r3, 48
 call memcpy
@@ -1528,57 +828,60 @@
 mov64 r1, r0
 lsh64 r1, 32
 rsh64 r1, 32
-jne r1, 26, +0x3120
+jne r1, 26, +0x1b00
 mov64 r6, 2
-ja +0x3140
-mov64 r9, 0
-jne r9, 0, +0x2f10
-mov64 r3, r1
-lddw r1, 0x300007fde
-jeq r2, 0, +0x2f28
-mov64 r1, r3
-lddw r2, 0x300000008
-jge r1, r2, +0x2f40
-mov64 r1, 1
-mov64 r2, 34
-call alloc::alloc::handle_alloc_error#decd
+ja +0x1b20
 mov64 r1, r0
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
 mov64 r6, r0
 mov64 r7, r1
 mov64 r1, r10
-add64 r1, -72
+add64 r1, -56
 call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 1]>#240e
 mov64 r0, r6
 mov64 r1, r7
-jne r0, 2, +0x908
+jne r0, 2, +0x430
 mov64 r0, 2
-ja +0x908
-call abort
+ja +0x430
+ldxdw r1, [r10-1028]
+stxdw [r10-136], r1
 mov64 r3, r10
-add64 r3, -333
-lddw r1, str("a Display implementation returned an error unexpectedly")
-mov64 r2, 55
-lddw r4, table[3:<core::fmt::Error as core::fmt::Debug>::fmt#aaa5]
-lddw r5, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/string.rs:2830:14)
+add64 r3, -136
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:162:23)
+call core::result::unwrap_failed#5a05
+ldxdw r1, [r10-1028]
+stxdw [r10-136], r1
+mov64 r3, r10
+add64 r3, -136
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:167:24)
 call core::result::unwrap_failed#5a05
 mov64 r3, r10
-add64 r3, -333
+add64 r3, -1
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:350:43)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:181:54)
 call core::result::unwrap_failed#5a05
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:185:77)
+call core::option::unwrap_failed#2d9b
 mov64 r3, r10
-add64 r3, -333
+add64 r3, -1
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:399:34)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:185:87)
 call core::result::unwrap_failed#5a05
-lddw r1, panic_location(programs/mayhem-program/src/state.rs:321:26)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:179:15)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:444:23)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:185:58)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:445:26)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:193:10)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:193:55)
+call core::panicking::panic_const::panic_const_sub_overflow#7310
